bin/node-template/runtime/src/template.rs

```diff
@@ -52,25 +56,52 @@ decl_module! {
 			Self::deposit_event(RawEvent::SomethingStored(something, who));
 			Ok(())
 		}
+
+		/// Another dummy entry point.
+		/// takes no parameters, attempts to increment storage value, and possibly throws an error
+		pub fn cause_error(origin) -> dispatch::DispatchResult {
+			// TODO: You only need this if you want to check it was signed.
+			let _who = ensure_signed(origin)?;
+
+			match Something::get() {
+				None => Err(Error::<T>::NoneValue)?,
+				Some(old) => {
+					let new = old.checked_add(1).ok_or(Error::<T>::StorageOverflow)?;
+					Something::put(new);
+					Ok(())
+				},
+			}
+		}
 	}
 }
 
+// The pallet's events
 decl_event!(
 	pub enum Event<T> where AccountId = <T as system::Trait>::AccountId {
-		// Just a dummy event.
-		// Event `Something` is declared with a parameter of the type `u32` and `AccountId`
-		// To emit this event, we call the deposit funtion, from our runtime funtions
+		/// Just a dummy event.
+		/// Event `Something` is declared with a parameter of the type `u32` and `AccountId`
+		/// To emit this event, we call the deposit funtion, from our runtime funtions
 		SomethingStored(u32, AccountId),
 	}
 );
 
-/// tests for this module
+// The pallet's errors
+decl_error! {
```

Comment:
better to put it above decl_module

Reply:
Okay, I'll do that. I'm just curious why that is preferred.